src/test/clojure/clara/test_rules.clj

```diff
@@ -907,6 +907,49 @@
            (set batch-inserts)
            (set single-inserts)))))
 
+(deftest test-accum-needing-token-partitions-correctly-on-fact-binding
+  (let [qhash (dsl/parse-query []
+                               [[WindSpeed (= ?ws windspeed)]
+                                [?ts <- (acc/all) :from [Temperature (= ?loc location)]]])
+        qfilter (dsl/parse-query []
+                                 [[WindSpeed (= ?ws windspeed)]
+                                  [?ts <- (acc/all) :from [Temperature (= ?loc location)
+                                                           (not (join-filter-equals temperature ?ws))]]])
+        t1mci (->Temperature 1 "MCI")
+        t2lax (->Temperature 2 "LAX")]
+
+    (doseq [[q join-type] [[qhash "hash join"]
+                           [qfilter "filter join"]]
+            :let [s (mk-session [q] :cache false)
+                  right-activates-before-left (-> s
+                                                  (insert t1mci
+                                                          t2lax)
+                                                  (insert (->WindSpeed 10 "MCI"))
+                                                  fire-rules
+                                                  (query q)
+                                                  frequencies)
+                  left-activates-before-right (-> s
+                                                  (insert (->WindSpeed 10 "MCI"))
+                                                  (insert t1mci
+                                                          t2lax)
+                                                  fire-rules
+                                                  (query q)
+                                                  frequencies)
```

Comment:
You're testing the left-activate of AccumulateNode fix here, but what about the mirror fix in left-retract of AccumulateNode?